### PR TITLE
Enable code coverage

### LINKED ISSUE #6009: Coveralls badge is pointing to old repo
As noted in https://github.com/apache/incubator-druid/pull/6007 the coveralls badge points to the old repo via https://coveralls.io/github/druid-io/druid

The new repo would be at https://coveralls.io/repos/apache/incubator-druid but there seem to be some missing settings somewhere. This ask is that the reason be investigated and rectified.

### PR DESCRIPTION
**Pull request by ccaominh**
Fixes #6009.

### Description

Code coverage was disabled via https://github.com/apache/incubator-druid/pull/3122 due to an issue with
cobertura in Travis CI. Switch code coverage tool from cobertura to
jacoco to avoid issue and re-enable coveralls for Travis CI.

<hr>

This PR has:
- [x] been self-reviewed.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -973,13 +973,33 @@
...
-                <artifactId>cobertura-maven-plugin</artifactId>
+                <groupId>org.jacoco</groupId>
+                <artifactId>jacoco-maven-plugin</artifactId>
+                <version>0.8.4</version>
+                <configuration>
+                    <excludes>
+                        <!-- Ignore generated code -->
+                        <exclude>org/apache/druid/math/expr/antlr/Expr*</exclude>  <!-- core -->
+                        <exclude>org/apache/druid/**/generated/*Benchmark*</exclude>  <!-- benchmarks -->
+                        <exclude>org/apache/druid/data/input/influx/InfluxLineProtocol*</exclude>  <!-- extensions-contrib/influx-extensions -->
```
[the remark this answers is not in the document]
Comment (ccaominh) [reply pom.xml]: Added benchmarks and integration-tests to the exclusions

[the remark this answers is not in the document]
Comment (clintropolis) [reply pom.xml]: Ah, it looks like you could add `org/testng/DruidTestRunnerFactory` too

Screen Shot 2019-08-20 at 12 48 44 PM

[the remark this answers is not in the document]
Comment (ccaominh) [reply pom.xml]: Added

**Comment by himanshug** [comment]
In https://github.com/apache/incubator-druid/pull/8359 coverage build failed due to drop by 0.09% (report says coverage of files not touched in the PR dropped)
also in any case, do we really want to fail the overall build on code coverage drop of 0.09% .

@clintropolis @gianm 
while code coverage is an interesting metric , I don't think it is a metric to chase for and fail the builds.

**Comment by ccaominh** [comment]
@himanshug I'm hoping https://github.com/apache/incubator-druid/pull/8374 will fix the issue with the fluctuating coverage reports

**Pull request merged**